lib/kitchen/document.rb

```diff
@@ -165,7 +165,32 @@ def create_element_from_string(string)
     def record_id_copied(id)
       return if id.blank?
 
-      @next_paste_count_for_id[id] ||= 1
+      @id_cp_count[id][:count]&.positive? ? @id_cp_count[id][:count] += 1 : @id_cp_count[id][:count] = 1
+      @id_cp_count[id][:last_paste] = false
+    end
+
+    # Keeps track that an element with the given ID has been cut.
+    #
+    # @param id [String]
+    #
+    def record_id_cut(id)
+      return if id.blank?
+
+      @id_cp_count[id][:count]&.positive? ? @id_cp_count[id][:count] -= 1 : @id_cp_count[id][:count]
+      @id_cp_count[id][:last_paste] = false
+    end
+
+    # Keeps track that an element with the given ID has been pasted.
+    #
+    # @param id [String]
+    #
+    def record_id_paste(id)
```

Comment:
to match the other methods, this should end `_pasted` (past tense).  Also `:last_paste` should be `:last_pasted`.